physics/doppler_frequency.py

```diff
@@ -0,0 +1,106 @@
+and frequency emitted from the source remain constant.
+
+All these results are summarized by the Doppler formula:
+
+    f = (f0 * (v + v0)) / (v - vs)
+
+where:
+    f: frequency of the wave
+    f0: frequency of the wave when the source is stationary
+    v: velocity of the wave in the medium
+    v0: velocity of the observer, positive if the observer is moving towards the source
+    vs: velocity of the source, positive if the source is moving towards the observer
+
+Doppler's effect has many applications in physics and engineering,
+such as radar, astronomy, medical imaging and seismology.
+
+References:
+https://en.wikipedia.org/wiki/Doppler_effect
+
+Now, we will implement a function that calculates the frequency of a wave as a function
+of the frequency of the wave when the source is stationary, the velocity of the wave
+in the medium, the velocity of the observer and the velocity of the source.
+"""
+
+
+def doppler_effect(
+    org_freq: float, wave_vel: float, obs_vel: float, src_vel: float
+) -> float:
+    """
+    Input Parameters:
+    -----------------
+    org_freq: frequency of the wave when the source is stationary
+    wave_vel: velocity of the wave in the medium
+    obs_vel: velocity of the observer, +ve if the observer is moving towards the source
+    src_vel: velocity of the source, +ve if the source is moving towards the observer
+
+    Returns:
+    --------
+    f: frequency of the wave as perceived by the observer
+
+    Docstring Tests:
+    >>> doppler_effect(100, 330, 10, 0) #observer moving towards the source
+    103.03030303030303
+    >>> doppler_effect(100, 330, -10, 0) #observer moving away from the source
+    96.96969696969697
+    >>> doppler_effect(100, 330, 0, 10) #source moving towards the observer
+    103.125
+    >>> doppler_effect(100, 330, 0, -10) #source moving away from the observer
+    97.05882352941177
+    >>> doppler_effect(100, 330, 10, 10) #observer and source moving towards each other
+    106.25
+    >>> doppler_effect(100, 330, -10, -10) #observer and source moving away
+    94.11764705882354
+    >>> doppler_effect(100, 330, 10, 330) #source moving at the same speed as the wave
+    Error: Division by zero
+    Infinite frequency implies vs = v and observer infront of the source
+    0.0
+    >>> doppler_effect(100, 330, 10, 340) #source moving faster than the wave
+    Error: Non-positive frequency
+    Non-positive frequency implies vs > v or v0 > v(in opposite direction)
+    0.0
+    >>> doppler_effect(100, 330, -340, 10) #observer moving faster than the wave
+    Error: Non-positive frequency
+    Non-positive frequency implies vs > v or v0 > v(in opposite direction)
+    0.0
+    """
+
+    try:
+        doppler_freq = (org_freq * (wave_vel + obs_vel)) / (wave_vel - src_vel)
+    except ZeroDivisionError:
+        print("Error: Division by zero")
+        print("Infinite frequency implies vs = v and observer infront of the source")
+        return 0.0
```

Comment:
`CONTRIBUTING.md` says that algorithmic functions should not `print()`.  Please let the ZeroDivisionError be raised to the caller because the answer is NOT 0.
```suggestion
        print("Error: Division by zero")
        print("Infinite frequency implies vs = v and observer infront of the source")
        return 0.0
```

Reply:
> `CONTRIBUTING.md` says that algorithmic functions should not `print()`. Please let the ZeroDivisionError be raised to the caller because the answer is NOT 0.

Yes, the print statement was somewhat redundant and not aligned with the Contributing Guidelines. Replaced it with raising the exception codes.